docs/DeveloperGuide.adoc

```diff
@@ -127,52 +128,53 @@ Given below is a quick overview of each component:
+For example, the <<Design-Logic,*`Logic`*>> component (see the class diagram given below) defines its <<API,API>> in the `Logic.java` interface and exposes its functionality using the `LogicManager.java` class.
 
 [discrete]
 ==== Events-Driven nature of the design
 
-The _Sequence Diagram_ below shows how the components interact for the scenario where the user issues the command `delete 1`.
+The _Sequence Diagram_ below (<<fig-ComponentInteraction>>) shows how the components interact in the scenario where the user issues the command *`delete 1`*.
 
-.Component interactions for `delete 1` command (part 1)
+[[fig-ComponentInteraction]]
+.Component interactions for *`delete 1`* command (part 1)
 image::SDforDeleteRecipe.png[width="800"]
 
 [NOTE]
-Note how the `Model` simply raises a `RecipeBookChangedEvent` when the Recipe Book data are changed, instead of asking the `Storage` to save the updates to the hard disk.
+Note how the <<Design-Model,*`Model`*>> simply raises a `_RecipeBookChangedEvent_` when the Recipe Book data are changed, instead of asking the <<Design-Storage,*`Storage`*>> to save the updates to the hard disk.
 
-The diagram below shows how the `EventsCenter` reacts to that event, which eventually results in the updates being saved to the hard disk and the status bar of the UI being updated to reflect the 'Last Updated' time.
+The diagram below shows how the `_EventsCenter_` reacts to that event, which eventually results in the updates being saved to the hard disk and the status bar of the UI being updated to reflect the 'Last Updated' time.
 
-.Component interactions for `delete 1` command (part 2)
+.Component interactions for *`delete 1`* command (part 2)
 image::SDforDeleteRecipeEventHandling.png[width="800"]
 
 [NOTE]
-Note how the event is propagated through the `EventsCenter` to the `Storage` and `UI` without `Model` having to be coupled to either of them. This is an example of how this Event Driven approach helps us reduce direct coupling between components.
+Note how the event is propagated through the `_EventsCenter_` to the <<Design-Storage,*`Storage`*>> and <<Design-Ui,*`UI`*>> without <<Design-Model,*`Model`*>> having to be coupled to either of them. This is an example of how this Event Driven approach helps us reduce direct coupling between components.
 
 The sections below give more details of each component.
 
 [[Design-Main]]
 === Main component
-`Main` has only one class called link:{repoURL}/src/main/java/seedu/recipe/MainApp.java[`MainApp`].
+<<Design-Main,*`Main`*>> has only one class called link:{repoURL}/src/main/java/seedu/recipe/MainApp.java[`MainApp`].
 
-* At app launch: It initializes the components in the correct sequence, and connects them up with each other.
-* At shut down: It shuts down the components and invokes cleanup method where necessary.
+* At app launch: Initializes the components in the correct sequence, and connects them up with each other.
```

Comment:
No subject? How about changing line 162 to
+<<Design-Main,*`Main`*>> has only one class called link:{repoURL}/src/main/java/seedu/recipe/MainApp.java[`MainApp`]. This class